### ISSUE TITLE
requireValidUser validate function becomes undefined in 1.0.0

### ISSUE THREAD

**Issue by sean-stanley** [Urigo/angular-meteor-auth#12]
Hi,

Using `$auth.requireValidUser(fn)` from ui-router resolve (and console) results in an error in the $$Auth factory.

I've determined the source of the error to be here: 

```
$$Auth.$awaitUser = function(validate) {
    validate = validate ? this.$bindToContext(validate) : function() {return true};

    if (!_.isFunction(validate))
      throw Error('argument 1 must be a function');
```
`$awaitUser` is initially called with the correct function for validate however after running the ternary operation validate becomes set to undefined. Most likely this is actually an issue in $$Core, where this.$bindToContext is not properly defined or returning no function. 

About this.$bindToContext, in $$Auth, `this` is a scope object but has none of the bound Meteor functions attached to it. I wonder if that could be part of the issue?

**Comment by sean-stanley** [comment]
Just curious, is the `this.$bindToContext` even necessary? It doesn't seem to be when I look at the rest of the `$awaitUser` function. Oh and I can confirm from tests that `this.$bindToContext` is defined but is returning undefined.

**Comment by Ben305** [comment]
I'm seeing this, too.

**Comment by DAB0mB** [comment]
@sean-stanley I'm pretty sure pr #13 fixes that issue. Can you please try it and tell if it is fine?

**Comment by sean-stanley** [comment]
@DAB0mB Yes your pr fixes that particular issue.

**Comment by Urigo** [comment]
will be published tomorrow

**Comment by Urigo** [comment]
published as `1.0.1`.
